### PR TITLE
Add eXist-db.app v5.0.0-RC1

### PR DESCRIPTION
**Pull request by adamretter**
I am afraid that I do not have a Mac with a decent enough Internet connection to be able to double-check everything for your task list. Hopefully it is okay though. I based this on a previous PR https://github.com/Homebrew/homebrew-cask-versions/pull/3066 from @joewiz

### CODE REVIEW

**Comment by joewiz** [comment]
Thanks @adamretter. I've completed the required checks on my own machine and confirm they all pass with your PR:


After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-versions/pulls
[closed issues]: https://github.com/caskroom/homebrew-versions/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

File: Casks/exist-db5.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'exist-db5' do
```
Comment (commitay) [inline Casks/exist-db5.rb]: This isn't a versioned cask as this is a pre-release version. It should not be called `exist-db5`.

Please rename it to `exist-db-{beta|dev|rc}` or similar.

**Comment by commitay** [comment]
`exist-db` has a java caveat, is java needed for this?

```
  caveats do
    depends_on_java '8'
  end
```

File: Casks/exist-db5.rb (hunk 1 above)
Comment (commitay) [reply Casks/exist-db5.rb]: @vitorgalvao I'm not sure I understand the `db-rc` -> `db3`  rename in the [other PR](https://github.com/Homebrew/homebrew-cask-versions/pull/3066)?

Comment (joewiz) [reply Casks/exist-db5.rb]: So you'd like us to change it to `exist-db-rc`, correct? We're happy to handle this however is best. 

I'll just add that [the PR template](https://github.com/Homebrew/homebrew-cask-versions/blob/master/.github/PULL_REQUEST_TEMPLATE.md) links to the [Cask Token Reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md) document, which states that version numbers should be removed from the end of tokens:

> Remove from the end: version numbers or incremental release designations such as “alpha”, “beta”, or “release candidate”. 

While that guidance makes sense for the main homebrew-cask repo, it leaves contributors to the homebrew-cask-versions repo wondering how to indicate, well, *the version of the app* in the token! We're happy to follow your advice.

**Comment by joewiz** [comment]
@commitay Yes, java8 is still needed. Attached is a patch. Or we could submit a new PR - whatever's best.

[0001-Align-cask-with-exist-db.patch](https://github.com/Homebrew/homebrew-cask-versions/files/2104624/0001-Align-cask-with-exist-db-patch.txt)

**Comment by adamretter** [comment]
@joewiz I added your patch commit to this PR, to make things easier for the maintainers.

**Comment by commitay** [comment]
I've opened https://github.com/Homebrew/homebrew-cask-versions/pull/5825 for `exist-db-nightly`

**Comment by commitay** [comment]
> > Remove from the end: version numbers or incremental release designations such as “alpha”, “beta”, or “release candidate”. 

> While that guidance makes sense for the main homebrew-cask repo, it leaves contributors to the homebrew-cask-versions repo wondering how to indicate, well, *the version of the app* in the token!

Thanks for the suggestion @joewiz! I'll open a PR with a couple of examples specific to this repo similar to what we have in [`drivers`](https://github.com/Homebrew/homebrew-cask-drivers/blob/master/CONTRIBUTING.md#naming-driver-casks)

**Comment by commitay** [comment]
Thanks @adamretter and @joewiz!

**Comment by adamretter** [comment]
@commitay Thank you very much :-)

**Pull request merged**